api/invite.go

```diff
@@ -970,6 +970,119 @@ func (a *Api) SendTeamInvite(res http.ResponseWriter, req *http.Request, vars ma
 
 }
 
+// @Summary Send invitation to a paatient for joining a medical team
+// @Description  create a notification for the invitee and send him an email with the invitation
+// @ID hydrophone-api-SendTeamInvitePatient
+// @Accept  json
+// @Produce  json
+// @Success 200 {object} models.Confirmation "invite details"
+// @Failure 400 {object} status.Status "userId, teamId and isAdmin were not provided or the payload is missing/malformed"
+// @Failure 401 {object} status.Status "Authorization token is missing or does not provide sufficient privileges"
+// @Failure 403 {object} status.Status "Authorization token is invalid"
+// @Failure 409 {object} status.Status "user already has a pending or declined invite OR user is already part of the team"
+// @Failure 422 {object} status.Status "Error when sending the email (probably caused by the mailling service"
+// @Failure 500 {object} status.Status "Internal error while processing the invite, detailled error returned in the body"
+// @Router /confirm/send/team/invite/patient [post]
+// @security TidepoolAuth
+func (a *Api) SendTeamInvitePatient(res http.ResponseWriter, req *http.Request, vars map[string]string) {
```

Comment:
This function is almost a copy of SendTeamInvite. The only difference is the template type/name and the authorization logic.
Can we reuse the same function or at least factorize?